From d71df4de4d9769b8391c52b9e516c834d54335b7 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Thu, 11 Feb 2021 07:29:43 +0100
Subject: [PATCH 1/1] buildman: 'Thread' object has no attribute 'isAlive'

The isAlive() method was deprecated in Python 3.8 and has been removed in
Python 3.9. See https://bugs.python.org/issue37804. Use is_alive() instead.

Since Python 2.6 is_alive() has been a synonym for isAlive(). So there
should be no problems for users using elder Python 3 versions.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 tools/buildman/builder.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/buildman/builder.py b/tools/buildman/builder.py
index c93946842a..6f6d759329 100644
--- a/tools/buildman/builder.py
+++ b/tools/buildman/builder.py
@@ -1691,7 +1691,7 @@ class Builder:
         term = threading.Thread(target=self.queue.join)
         term.setDaemon(True)
         term.start()
-        while term.isAlive():
+        while term.is_alive():
             term.join(100)
 
         # Wait until we have processed all output
-- 
2.30.0

